examples/shared/mod.rs

```diff
@@ -72,18 +72,20 @@ pub async fn create_camera(
     );
 
     // Unwrap everything.
-    let skybox = SkyBox {
+    let skybox = SkyBoxBuilder {
         front: Some(front.unwrap()),
         back: Some(back.unwrap()),
         left: Some(left.unwrap()),
         right: Some(right.unwrap()),
         top: Some(top.unwrap()),
         bottom: Some(bottom.unwrap()),
-    };
+    }
+    .build()
+    .unwrap();
 
     // Set S and T coordinate wrap mode, ClampToEdge will remove any possible seams on edges
     // of the skybox.
-    for skybox_texture in skybox.textures().iter().filter_map(|t| t.clone()) {
+    for skybox_texture in skybox.cubemap() {
```

Comment:
Looks like you iterating over Option here, maybe `if let Some(cubemap) = skybox.cubemap() {` will be better here?

Reply:
Thanks for the review. You're right `if let `would be more convenient here.